src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,965 @@
+        }
+
+        return ofYearDay((int) year, (int) doy);
+    }
+
+    /**
+     * Consistency check for dates manipulations after calls to
+     *   {@link #plus(long, TemporalUnit)},
+     *   {@link #minus(long, TemporalUnit)},
+     *   {@link #until(AbstractDate, TemporalUnit)} or
+     *   {@link #with(TemporalField, long)}.
+     *
+     * @param prolepticYear the International fixed proleptic-year
+     * @param month         the International fixed month, from -1 to 13 (-1 for Leap Day, 0 for Year Day)
+     * @param day           the International fixed day-of-month, from -1 to 28 (-1 for Leap Day, 0 for Year Day)
+     * @return
+     */
+    private static InternationalFixedDate resolvePreviousValid(final int prolepticYear, final int month, final int day) {
+        if (month == 0 && day == 0) {
+            return createYearDay(prolepticYear);
+        }
+
+        if (month == -1 && day == -1 && InternationalFixedChronology.INSTANCE.isLeapYear(prolepticYear)) {
+            return createLeapDay(prolepticYear);
+        }
+
+        int monthR = month == -1 ? 7
+                                 : month == 0 ? InternationalFixedChronology.MONTHS_IN_YEAR
+                                              : Math.min(month, InternationalFixedChronology.MONTHS_IN_YEAR);
+        int dayR = day == -1 ? 1
+                             : day == 0 ? InternationalFixedChronology.DAYS_IN_MONTH
+                                        : Math.min(day, InternationalFixedChronology.DAYS_IN_MONTH);
+
+        return of(prolepticYear, monthR, dayR);
+    }
+
+    //-----------------------------------------------------------------------
+    /**
+     * Factory method, validates the given triplet year, month and dayOfMonth.
+     * Special values are required for Year Day (N/0/0) and Leap Day (N/-1/-1).
+     *
+     * @param prolepticYear the International fixed proleptic-year
+     * @param month         the International fixed month, from -1 to 13 (-1 for Leap Day, 0 for Year Day)
+     * @param dayOfMonth    the International fixed day-of-month, from -1 to 28 (-1 for Leap Day, 0 for Year Day)
+     * @return the International fixed date
+     * @throws DateTimeException if the date is invalid
+     */
+    static InternationalFixedDate create(final int prolepticYear, final int month, final int dayOfMonth) {
+        InternationalFixedChronology.YEAR_RANGE.checkValidValue(prolepticYear, ChronoField.YEAR_OF_ERA);
+        InternationalFixedChronology.MONTH_OF_YEAR_RANGE.checkValidValue(month, ChronoField.MONTH_OF_YEAR);
+        InternationalFixedChronology.DAY_OF_MONTH_RANGE.checkValidValue(dayOfMonth, ChronoField.DAY_OF_MONTH);
+
+        if (((month < 1 || dayOfMonth < 1) && (dayOfMonth != month)) ||
+            ((month == -1) && (dayOfMonth == -1) && !InternationalFixedChronology.INSTANCE.isLeapYear(prolepticYear))) {
+            throw new DateTimeException("Invalid date: " + prolepticYear + '/' + month + '/' + dayOfMonth);
```

Comment:
`create(...)` - need to spit out different error for erroneous leap-year status (see existing implementations).